### PR TITLE
Update style_guide for Hugo/markdown

### PR DESCRIPTION
**Pull request by IanMadd**
Signed-off-by: IanMadd <imaddaus@chef.io>

### Description

This rewrites the Style Guide for Markdown and Hugo.

### Check List

- [ ] Spell Check
- [X] Local build
- [X] Examine the local build
- [ ] All tests pass

### CODE REVIEW

File: content/style_guide.md [hunk 1]
```diff
@@ -16,41 +16,74 @@ aliases = "/style_guide.html"
...
-    =====================================================
+menu title
+----------------
+The title of the page as it would appear in the left navigation menu. 
+
+menu identifier 
+----------------
+The identifier of the page that you are writing. Each identifier must be unique. 
+
+The convention we've adopted is to use the identifier of the page's parent, followed by a forward slash, then the page file, a space, and then the page title.
```
Comment (kagarmoe) [inline content/style_guide.md]: An example might be helpful

File: content/style_guide.md [hunk 2]
```diff
@@ -130,202 +162,188 @@ Bulleted Lists
...
 
-    #. text goes here
-    #. text goes here
-    #. text goes here
-    #. text goes here
-
-Use the number symbol (\#) to let Sphinx handle the actual ordering. If
-the number list needs to change later, you don't have to worry about
-making sure the numbers are in the correct order.
+    1. text goes here
```
Comment (kagarmoe) [inline content/style_guide.md:183]: Can we use:
1.
1.
1.

& Hugo generates the numbers? That is preferable to maintaining numbered lists.

File: content/style_guide.md [hunk 3]
```diff
@@ -130,202 +162,188 @@ Bulleted Lists
...
+      </tr>
+      <tr>
+        <td>Infra Server</td>
+        <td>Great</td>
+      </tr>
+      <tr>
+        <td>Habitat</td>
+        <td>Super cool</td>
+      </tr>
+    </table>
```
Comment (kagarmoe) [inline content/style_guide.md]: Even though Hugo can handle HTML tables, I'd prefer to limit us to markdown tables in the docs, in order to keep the pages as single-language as possible.

File: content/style_guide.md [hunk 4]
```diff
@@ -130,202 +162,188 @@ Bulleted Lists
...
-An external link points to something that does not reside within
-[docs.chef.io](https://docs.chef.io). An external link requires an HTTP
-address. In general, it's better to spell out the HTTP address fully, in
-case the topic is printed out:
+will produce this: 
 
-> `http://www.codecademy.com/tracks/ruby`
+[Ruby Programming Language](https://www.ruby-lang.org/)
+
+External links requires an HTTP address.
```
Comment (kagarmoe) [inline content/style_guide.md:249]:
```suggestion
External links require an HTTP address.
```

File: content/style_guide.md [hunk 5]
```diff
@@ -130,202 +162,188 @@ Bulleted Lists
...
-       }
+    ```javascript
+    {
+      "id": "charlie",
+      "uid": 1005,
+      "gid":"ops",
+      "shell":"/bin/zsh",
+      "comment":"Crazy Charlie"
+    }
+    ```
```
Comment (kagarmoe) [inline content/style_guide.md:289]: Does Hugo support code blocks for `powershell`  and `=`output `?

File: content/style_guide.md [hunk 6]
```diff
@@ -130,202 +162,188 @@ Bulleted Lists
...
+    with more
+    text as required to
+    complete the block of text.
+    end.
+    
+```
 
-Chef docs uses tags to indicate text that is used in more than one
-topic:
+Shortcodes
```
Comment (kagarmoe) [inline content/style_guide.md]: Perhaps we should call this "Reusing text" or something similar? Shortcodes have different uses (as you well know :-) ) and we don't want to mislead our readers.

File: content/style_guide.md [hunk 7]
```diff
@@ -334,69 +352,81 @@ In general, notes and warnings are not the best way to present important
...
+Images require an exclamation point, square brackets around the alt text, parenthesis around the path to the image file starting in the `static` directory, and then optional hover text:
 
-    .. image:: ../../images/icon_chef_client.svg
-       :width: 100px
-       :align: center
+```md
+![Alt Text](/path/to/img.png "Optional Title")
+```
 
 Images should be 96 dpi and no larger than 600 pixels wide. Ideally, no
```
Comment (kagarmoe) [inline content/style_guide.md]: We should choose between 600 & 550px width. We should also figure out how to lint image widths.

File: content/style_guide.md [hunk 8]
```diff
@@ -456,121 +480,3 @@ For Chef applications and components, use:
 -   `Chef Client` for the Chef client up to version 14.x
 -   `Chef Infra Client` for Chef Infra Client version 15 and above
 -   `Chef Automate` for the Chef Automate product
```
Comment (kagarmoe) [inline content/style_guide.md]: Chef Automate
Chef Habitat
Chef Infra (formerly Chef)
Chef Infra Client (Use Chef Client up to version 14.x)
Chef Infra Server (Formerly Chef Server)
Chef InSpec

File: content/style_guide.md [hunk 9]
```diff
@@ -456,121 +480,3 @@ For Chef applications and components, use:
...
-
-TOC Trees
```
Comment (kagarmoe) [inline content/style_guide.md:465]: Do we need to add pages manually to the sitemap.xml or the index.html?

**Review by kagarmoe — CHANGES_REQUESTED** [review]

File: content/style_guide.md (hunk 9 above)
Comment (IanMadd) [reply content/style_guide.md:465]: There is no sitemap, but we do have pages listed in the _index.md.

File: content/style_guide.md (hunk 6 above)
Comment (IanMadd) [reply content/style_guide.md]: "Recurring text blocks" or "Repeating text blocks"?

File: content/style_guide.md (hunk 2 above)
Comment (IanMadd) [reply content/style_guide.md:183]: yup

File: content/style_guide.md (hunk 5 above)
Comment (IanMadd) [reply content/style_guide.md:289]: Here's the docs on [syntax highlighting](https://gohugo.io/content-management/syntax-highlighting/)
So, yes on powershell.
I'm not sure how you want an output to look, but there is an option to highlight certain lines with a different color using the `hi_lines` option.

File: content/style_guide.md (hunk 7 above)
Comment (IanMadd) [reply content/style_guide.md]: I haven't found a way in Hugo to lint images. Maybe we could add something to the makefile?

File: content/style_guide.md (hunk 6 above)
Comment (IanMadd) [reply content/style_guide.md]: I replaced this with `Repeating Text Blocks`

File: content/style_guide.md (hunk 3 above)
Comment (IanMadd) [reply content/style_guide.md]: I removed the html tables.

**Review by kagarmoe — APPROVED** [review]
Resolve the merge conflict & you're good to go!

**Review by schisamo — APPROVED** [review]

**Pull request merged**